store.go

```diff
@@ -701,6 +713,25 @@ func (s *store) load() error {
 	return nil
 }
 
+func (s *store) LockDigest(d digest.Digest) error {
+	lockfile, err := GetLockfile(filepath.Join(s.lockRoot, d.Encoded()))
+	if err != nil {
+		return err
+	}
+	lockfile.Lock()
+	return nil
+}
+
+func (s *store) UnlockDigest(d digest.Digest) error {
+	lockPath := filepath.Join(s.lockRoot, d.Encoded())
+	lockfile, err := GetLockfile(lockPath)
```

Comment:
Nalin pointed out in 
https://github.com/containers/image/pull/607#discussion_r267525606 that every call to `GetLockfile` allocates a FD that is never closed.

Reply:
… allocating only one FD per path, sorry.

Reply:
This is now being addressed in the latest commit:
```
Add a method to the store to generate a lock file for a specific digest.
Such a digest-specific lock file is needed to synchronize threads and
processes when copying blobs from a registry to the containers-storage.

Whenever a layer is about to get copied, the lock must be acquired which
indicates to other processes and threads that the layer/blob is already
being copied.

To avoid leaking file descriptors, add a new Close() method to the
Locker.  We are now counting references on locks, which are being
decremented in Close().  When the last reference called Close(),
the lock file's FD is being closed as well.  The reference counter is
incremented for each new invocation on Get{RO,}Lockfile().
```

Reply:
In short: we're now reference counting the lock files and have a `Locker.Close()` method which will close the file descriptor when the reference counter is 0.

The store now just returns a locker, while the storage image destination must take care of book keeping of the Locker objects.

Reply:
Actually, it may just be much cleaner to open and close the file on the fly. We're alreay refcounting the with each Lock() and Unlock(). Github turns out to be a good rubbderduck.